src/Compilers/CSharp/Portable/Symbols/Metadata/PE/PEEventSymbol.cs

```diff
@@ -92,11 +92,16 @@ private enum Flags : byte
                 _eventType = metadataDecoder.GetTypeOfToken(eventType);
             }
 
+            TypeSymbol originalEventType = _eventType;
+            _eventType = DynamicTypeDecoder.TransformType(_eventType, 0, handle, moduleSymbol);
```

Comment:
Consider adjusting the code so that only the final value was stored in the ```_eventType``` field.